enterprise/models.py

```diff
@@ -807,6 +830,39 @@ def consent_needed(self):
         enterprise_customer = self.enterprise_customer_user.enterprise_customer
         return enterprise_customer.enforces_data_sharing_consent(EnterpriseCustomer.AT_ENROLLMENT)
 
+    @property
+    def audit_reporting_disabled(self):
+        """
+        Specify whether audit track data reporting is disabled for this enrollment.
+
+        * If the enterprise customer associated with this enrollment enables audit track data reporting,
+          simply return False.
+        * If the enterprise customer associated with this enrollment does not enable audit track data reporting,
+          return True if we are dealing with an audit enrollment, and False otherwise.
+
+        :return: True if audit track data reporting is disabled, False otherwise.
+        """
+        if not self.enterprise_customer_user.enterprise_customer.enables_audit_data_reporting:
+            return self.is_audit_enrollment
+
+        # Since audit data reporting is enabled, we always return False here.
+        return False
+
+    @property
+    def is_audit_enrollment(self):
+        """
+        Specify whether the course enrollment associated with this ``EnterpriseCourseEnrollment`` is in audit mode.
+
+        :return: Whether the course enrollment mode is of an audit type.
+        """
+        course_enrollment_api = EnrollmentApiClient()
+        course_enrollment = course_enrollment_api.get_course_enrollment(
+            self.enterprise_customer_user.username,
+            self.course_id
+        )
+        audit_modes = getattr(settings, 'ENTERPRISE_COURSE_ENROLLMENT_AUDIT_MODES', ['audit', 'honor'])
+        return course_enrollment.get('mode') in audit_modes
```

Comment:
@brittneyexline Thanks so much for testing this! I understand how you were able to reproduce this, but yeah, this is something we can consider another time. For now, I'll change it to take care of the NoneType case.